### ISSUE TITLE
"TypeError: Method "toString" called on incompatible object."

### ISSUE THREAD

**Issue by hayds** [thewhitespace/Xplore#8]
Hi i'm using version 4.4 on Jakarta Patch 3. I get the following error 

"TypeError: Method "toString" called on incompatible object."
--

When I run the following in Xplore (Server/Global)

var a = new KBVersioning();
a

**Comment by jneale** [comment]
Not sure what plugin KBVersioning comes with, so I can't test.
The incompatible object error happens when you try to work with Java objects that do not have toString functions.
Click the cog icon and turn off "Show String Values" and that should allow you to see the contents (if they are not hidden).

**Comment by hayds** [comment]
Thanks James, that did work. Would love to know the name of the object that caused the error and the ability to add it to the blacklist if possible. Any way you can report the object name at exception time?

Thanks

**Comment by jneale** [comment]
Unfortunately not - wish I could! It's a long standing issue with ServiceNow using the Rhino engine to front Java code: some exceptions just get lost with no output. It's one of the things that's made developing and maintaining Xplore difficult.

You could try running a for-in loop in Xplore on the object and printing the contents and testing toString on each property.

```for (var x in KBVersioning) {
    gs.addInfoMessage(x + ' >> "' + KBVersioning[x] + '"');
}```

**Comment by hayds** [comment]
Funny you should mention... that's exactly what I am doing now :D

I'm looping and running a try catch and outputting the current index in the
catch.

Thanks

**Comment by hayds** [comment]
Found the latest culprit....

new KBViewModel().kbKnowledgeSI.toString();

**Comment by hayds** [comment]
Hi James, turns out  "new KBViewModel().kbKnowledgeSI.toString();" was not
actually causing my issue after all.

Using Xplore 4.4 on Jakarta if I type the following in to snd_xplore.do
page in the script section...

new KBViewModel();

I will get...  "TypeError: Method "toString" called on incompatible object."

But if I go add a new system property called "snd_xplore.blacklist" and add
the value "[Target]" and try again, I wont get the error.

I don't have the time right this second to understand why, but wanted to
let you know. I think its maybe calling toString() on the target object
itself and in this case it doesnt have a toString method or cant handle it.

Interestingly running

var a = new KBViewModel();
gs.print(typeof a);

in a normal background script (sys.scripts.do) outputs "*** Script: object"

**Comment by jneale** [comment]
This turned out to be an issue in the code - likely caused by a change in the backend as I haven't come across it until recently. I've added a fix in for it in v4.5.
Thanks!
